File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int p1 = 0;
        int p2 = nums.length - 1;

        while(p1<p2) {
            if(nums[p2] + nums[p1] > target) {
                p2--;
            } else if (nums[p2] + nums[p1] < target) {
                p1++;
            } else {
                break;
            }
        }
        int[] res = {p1+1, p2+1};
        return res;
    }
}

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        def helper(i,curr):
            if i == len(nums):
                return [[]]
            
            res = []
            perms = helper(i+1,curr)
            for p in perms:
                for j in range(len(p)+1):
                    pCopy = p.copy()
                    pCopy.insert(j,nums[i])
                    res.append(pCopy)
            return res
        perms = [[]]
        for n in nums:
            newPerms = []
            for p in perms:
                for j in range(len(p)+1): # Here we want to add to every possible position
                    pCopy = p.copy()
                    pCopy.insert(j,n)
                    newPerms.append(pCopy)
            perms = newPerms
        return perms
                    
        # return helper(0,[])